driver-core/src/main/java/com/datastax/driver/core/NettyCustomizer.java

```diff
@@ -0,0 +1,165 @@
+ * compile and/or runtime errors.</strong>
+ *
+ * @jira_ticket JAVA-640
+ * @since 2.0.10
+ */
+public class NettyCustomizer {
+
+    /**
+     * The default instance of NettyCustomizer to use.
+     */
+    public static final NettyCustomizer DEFAULT_INSTANCE = new NettyCustomizer();
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NettyCustomizer.class);
+
+    private static final class EventLoopGroupFactory {
+
+        private static final Class<? extends EventLoopGroup> EVENT_LOOP_GROUP_CLASS;
+
+        private static final Constructor<? extends EventLoopGroup> EVENT_LOOP_GROUP_CONSTRUCTOR;
+
+        private static final Class[] EVENT_GROUP_ARGUMENTS = { int.class, ThreadFactory.class };
+
+        static {
+            boolean useEpoll = NettyUtil.isEpollAvailable();
+            EVENT_LOOP_GROUP_CLASS = useEpoll ? EpollEventLoopGroup.class : NioEventLoopGroup.class;
+            Constructor<? extends EventLoopGroup> constructor = null;
+            try {
+                constructor = EVENT_LOOP_GROUP_CLASS.getDeclaredConstructor(EVENT_GROUP_ARGUMENTS);
+            } catch (NoSuchMethodException e) {
+                LOGGER.warn("Could not resolve {}(int, ThreadFactory), using default constructor instead.",
+                    EVENT_LOOP_GROUP_CLASS.getName());
+            }
+            EVENT_LOOP_GROUP_CONSTRUCTOR = constructor;
+        }
+
+        public static EventLoopGroup newInstance(ThreadFactory factory) {
+            try {
+                // Use default constructor if the specialized constructor could not be resolved.
+                EventLoopGroup eventLoopGroup;
+                if (EVENT_LOOP_GROUP_CONSTRUCTOR != null) {
+                    eventLoopGroup = EVENT_LOOP_GROUP_CONSTRUCTOR.newInstance(0, factory);
+                } else {
+                    eventLoopGroup = EVENT_LOOP_GROUP_CLASS.newInstance();
+                }
+                return eventLoopGroup;
+            } catch (Exception e) {
+                throw new AssertionError("Could not create an instance of "
+                    + EVENT_LOOP_GROUP_CLASS.getName()
+                    + ", this should not happen");
+            }
+        }
+
+    }
+
+    /**
+     * Return the {@link EventLoopGroup} instance to use.
+     *
+     * @return the {@link EventLoopGroup} instance to use.
+     */
+    public EventLoopGroup createEventLoopGroup(ThreadFactory threadFactory) {
+        return EventLoopGroupFactory.newInstance(threadFactory);
+    }
+
+    /**
+     * Return the specific {@link SocketChannel} subclass to use.
+     * The default implementation returns {@link EpollSocketChannel} if {@link NettyUtil#isEpollAvailable()}  epoll is available},
+     * or {@link NioSocketChannel} otherwise.
+     *
+     * @return The {@link SocketChannel} subclass to use.
+     */
+    public Class<? extends SocketChannel> channelClass() {
+        return NettyUtil.isEpollAvailable() ? EpollSocketChannel.class : NioSocketChannel.class;
```

Comment:
Also all the static initialization in NettyUtil only happens if it gets called, so clients overriding this method will not have it executed.